internal/sdkprovider/service/flink/flink_application_version.go

```diff
@@ -186,13 +189,40 @@ func resourceFlinkApplicationVersionCreate(ctx context.Context, d *schema.Resour
 		sinks = expandFlinkApplicationVersionSourcesOrSinks(d.Get("sink").(*schema.Set).List())
 	}
 
-	r, err := client.FlinkApplicationVersions.Create(ctx, project, serviceName, applicationID, aiven.GenericFlinkApplicationVersionRequest{
-		Statement: d.Get("statement").(string),
-		Sources:   sources,
-		Sinks:     sinks,
-	})
-	if err != nil {
-		return diag.Errorf("cannot create Flink Application Version: %+v - %v", expandFlinkApplicationVersionSourcesOrSinks(d.Get("sources").(*schema.Set).List()), err)
+	var r *aiven.DetailedFlinkApplicationVersionResponse
+
+	if err := retry.RetryContext(ctx, time.Second*30, func() *retry.RetryError {
+		var err error
+
+		r, err = client.FlinkApplicationVersions.Create(
+			ctx,
+			project,
+			serviceName,
+			applicationID,
+			aiven.GenericFlinkApplicationVersionRequest{
+				Statement: d.Get("statement").(string),
+				Sources:   sources,
+				Sinks:     sinks,
+			},
+		)
+		if err != nil {
+			return &retry.RetryError{
+				Err: err,
+				Retryable: regexp.MustCompile(
+					"Integration not found: " +
```

Comment:
Perhaps we can simplify this like `regexp.MustCompile("Integration not found:")`, whatever goes after service integration is unimportant. 

For example, if someone alters an error message on the back-end, like adding a dot in the end or something else, that would impact us here.

Reply:
at least we'll know then if something changed around this API :)